libr/io/section.c

```diff
@@ -57,7 +57,10 @@ static RIOSection *findMatching (RIO *io, ut64 paddr, ut64 vaddr, ut64 size, ut6
 R_API RIOSection *r_io_section_add(RIO *io, ut64 offset, ut64 vaddr, ut64 size, ut64 vsize, int rwx, const char *name, ut32 bin_id, int fd) {
 	int update = 0;
 	RIOSection *s;
-	if (size == 0 || size > 0xf0000000) {
+	ut64 curr = lseek (fd, 0, SEEK_CUR);
```

Comment:
for me, in either way it would be same. Didn't understand the major difference in what should be preferred though :( But let's go for r_io_desc_size :) Or should I do put it this way: check if io->desc exists, if yes, go for r_io_desc_size else r_io_size. Again not sure if io->desc will always be set or not so this check would be safer for now.